HARK/core.py

```diff
@@ -16,6 +16,101 @@
 from .parallel import multi_thread_commands, multi_thread_commands_fake
 from warnings import warn
 
+# pyflakes kept flagging _log commands in core.py as undefined unless the log stuff
+# was here at the beginning (and not just in __init__.py
+
+import logging
+
+logging.basicConfig(format="%(message)s")
+
+_log = logging.getLogger("HARK")
+
+_log.setLevel(logging.ERROR)
+
+
+def disable_logging():
+    _log.disabled = True
+
+
+def enable_logging():
+    _log.disabled = False
+
+
+def warnings():
+    _log.setLevel(logging.WARNING)
+
+
+def quiet():
+    _log.setLevel(logging.ERROR)
+
+
+def verbose():
+    _log.setLevel(logging.INFO)
+
+
+def set_verbosity_level(level):
+    _log.setLevel(level)
+
+
+def core_check_condition(name, test, messages, messaging_level, verbose_messages, fact, stage_solution, quietly=False):
+    """
+    Checks whether parameter values of a model satisfy a condition
+
+    Parameters
+    ----------
+    name : string
+         Name for the condition.
+
+    test : function(self -> boolean)
+         A function (of self) which tests the condition
+
+    verbose : Boolean
+         If False, print minimal information about the condition and exit
+         If True, print more detailed information and a reference
+
+    messages : dict{boolean : string}
+        A dictionary with boolean keys containing values
+        for messages to print if the condition is
+        true or false.
+
+    messaging_level : int
+        Controls verbosity of messages. logging.DEBUG is most verbose,
+        logging.INFO is less verbose, logging.WARNING speaks up only if the
+        model might have problems, logging.CRITICAL indicates it is degenerate.
+
+    verbose_messages : dict{boolean : string}
+        (Optional) A dictionary with boolean keys containing supplemental
+        messages to print if the condition is
+        true or false if messaging_level is logging.DEBUG
+
+    fact : string
+         Name of the fact (for recording the results)
+
+    stage_solution : solution for a stage of the problem containing a condition 
+        to be tested.  Must have dict 'conditions' [name]
+
+    quiet : boolean
+        If True, construct the conditions but print nothing
```

Comment:
I don't understand why there needs to be both a verbosity and a quietness setting.